src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -750,6 +750,23 @@ private void InitializeCustomNodeManager()
                 if (customNodeSearchRegistry.Contains(info.FunctionId))
                     return;
 
+                var elements = SearchModel.SearchEntries.OfType<CustomNodeSearchElement>().Where(
+                                x =>
+                                {
+                                    return string.Compare(x.Path, info.Path, StringComparison.OrdinalIgnoreCase) == 0;
+                                }).ToList();
+
+                if (elements.Any())
+                {
+                    foreach (var element in elements)
+                    {
+                        customNodeSearchRegistry.Remove(element.ID);
+                        element.SyncWithCustomNodeInfo(info);
+                        SearchModel.Update(element);
+                    }
+                    return;
+                }
```

Comment:
Nice! Since it is an update operation, we do not need to remove the element ID from customNodeSearchRegistry.